control-plane/consul/resource_client.go

```diff
@@ -32,9 +19,9 @@ func NewResourceServiceClient(ctx context.Context, config discovery.Config, logg
 	state, err := watcher.State()
 	if err != nil {
 		watcher.Stop()
```

Comment:
~ Do we need to `Stop()` here since the `watcher` is managed outside this file? Might be cleaner just to return the error and let the caller deal w/ shutdown

Reply:
You are right! I forgot to change this after the re-factor.